### PR TITLE
Adding 7 new integrity checks

### PR DESCRIPTION
**Pull request by danielduhh**
### Description:
This brings in the following new integrity checks:

1. **BigNodeBadDataCheck** - Identify complex intersections that have too many paths and junction edges.
2. **ConnectivityCheck** - Identify nodes that should be connected to a nearby node or edge.
3. **ConflictingTagCombinationCheck** - The purpose of this check is to identify contradicting or incompatible tags (building, natural, and route=FERRY) on Atlas Objects, as well as to verify that values are appropriate when a service tag is used.
4. **ImproperAndUnknownRoadNameCheck** - This check flags improper road name values. Road names are considered improper if they match a road type, or a road surface material, as defined in the Highway, Bridge, and Surface Tags. Additionally, improper roads names are listed in the names.improper configuration value.
5. **DuplicateLocationInPolyLineCheck** - Flags repeating locations in a PolyLine
6. **InconsistentRoadClassificationCheck** - This identifies roads, that transition from one classification to another and then back to the original classification.
7. **ShortNameCheck** - Identifies any and all names within a name tag that do not contain at least two characters.

### Potential Impact:

No downstream impact. Just more available checks to run.

### Unit Test Approach:

Unit tests added for each check

### Test Results:

Pass as expected

### CODE REVIEW

File: src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingTagCombinationCheckTestRule.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Area.Known;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Edge;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Line;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Loc;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Node;
+
+/**
+ * Data generator for {@link ConflictingTagCombinationCheckTest}
+ *
+ * @author mkalender <<<<<<< HEAD
```
Comment (Bentleysb) [inline src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingTagCombinationCheckTestRule.java]: Old merge marker

File: src/test/resources/org/openstreetmap/atlas/checks/validation/intersections/MEX_7-24-55-subatlas.txt [hunk 2]
```diff
@@ -0,0 +1,2459 @@
+# Nodes
```
Comment (seancoulter) [inline src/test/resources/org/openstreetmap/atlas/checks/validation/intersections/MEX_7-24-55-subatlas.txt]: Where's this file used?

Comment (danielduhh) [reply src/test/resources/org/openstreetmap/atlas/checks/validation/intersections/MEX_7-24-55-subatlas.txt]: https://github.com/osmlab/atlas-checks/pull/217/files#diff-3e0a2f4f82c8d8e8c1d457faa5961d42

Comment (danielduhh) [reply src/test/resources/org/openstreetmap/atlas/checks/validation/intersections/MEX_7-24-55-subatlas.txt]: BigNodeBadDataCheckTestRule

**Pull request merged**